### PR TITLE
Feat/powerpoint conversion

### PR DESCRIPTION
**Pull request by jgoldhammer**
This PR adds the possibility to convert files with powerpoint.
All tests are green now - converting ppt to pptx, ppt to pdf and pptx to pdf works now.

Notes:
As MS Office is a little bit special to be used for automation, powerpoint causes more headaches.
- Powerpoint must run in foreground (in background the started instance cannot be found via GetObject)
- Powerpoint has no reliable method to be closed by vbs script, so that we need to use taskkill from vbsscript

### CODE REVIEW

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java [hunk 1]
```diff
@@ -88,7 +97,8 @@ public String toString() {
     }
 
     /**
-     * A holder type for type and subtype names of known {@link com.documents4j.api.DocumentType}s.
+     * A holder type for type and subtype names of known
```
Comment (raphw) [inline documents4j-api/src/main/java/com/documents4j/api/DocumentType.java:100]: Please undo the formatting change.

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java [hunk 2]
```diff
@@ -66,8 +73,10 @@ public String getSubtype() {
 
     @Override
     public boolean equals(Object other) {
-        if (this == other) return true;
-        if (other == null || getClass() != other.getClass()) return false;
```
Comment (raphw) [inline documents4j-api/src/main/java/com/documents4j/api/DocumentType.java:70]: Please undo the formatting change.

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java [hunk 3]
```diff
@@ -44,7 +49,9 @@ public DocumentType(String type, String subtype) {
     /**
      * Creates a new document type.
      *
-     * @param fullType The MIME type's type name and subtype name, separated by a {@code /}.
```
Comment (raphw) [inline documents4j-api/src/main/java/com/documents4j/api/DocumentType.java:47]: Please undo the formatting change.

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java [hunk 4]
```diff
@@ -30,8 +33,10 @@
     /**
      * Creates a new document type.
      *
-     * @param type    The MIME type's type name.
-     * @param subtype The MIME type's subtype name.
+     * @param type
```
Comment (raphw) [inline documents4j-api/src/main/java/com/documents4j/api/DocumentType.java:36]: Please undo the formatting change.

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java [hunk 5]
```diff
@@ -3,10 +3,10 @@
 import java.io.Serializable;
 
 /**
- * Represents an immutable document <a href="http://en.wikipedia.org/wiki/Internet_media_type">MIME</a> type.
```
Comment (raphw) [inline documents4j-api/src/main/java/com/documents4j/api/DocumentType.java:6]: Please undo the formatting change.

File: documents4j-transformer-msoffice/documents4j-transformer-msoffice-base/src/main/java/com/documents4j/conversion/msoffice/AbstractMicrosoftOfficeBridge.java [hunk 6]
```diff
@@ -64,15 +64,20 @@ protected StartedProcess doStartConversion(File source, DocumentType sourceType,
         try {
             MicrosoftOfficeFormat microsoftOfficeFormat = formatOf(targetType);
             // Always call destroyOnExit before adding a listener: https://github.com/zeroturnaround/zt-exec/issues/14
+            String[] command = {"cmd", "/S", "/C",
+                    doubleQuote(conversionScript.getAbsolutePath(),
+                            source.getAbsolutePath(),
+                            target.getAbsolutePath(),
+                            microsoftOfficeFormat.getValue())};
+
+            getLogger().trace("Running command for conversion " + command.toString());
```
Comment (raphw) [inline documents4j-transformer-msoffice/documents4j-transformer-msoffice-base/src/main/java/com/documents4j/conversion/msoffice/AbstractMicrosoftOfficeBridge.java:73]: This won't log anything meaningful. Please use the log placeholder notation *Running command for conversion {}* and wrap the array in `Arrays.toString()` to have the log print something.

File: documents4j-transformer/src/main/java/com/documents4j/conversion/ExternalConverterDiscovery.java [hunk 7]
```diff
@@ -11,43 +11,42 @@
 
 enum ExternalConverterDiscovery {
 
-    MICROSOFT_WORD("com.documents4j.conversion.msoffice.MicrosoftWordBridge"),
-    MICROSOFT_EXCEL("com.documents4j.conversion.msoffice.MicrosoftExcelBridge");
+    MICROSOFT_WORD("com.documents4j.conversion.msoffice.MicrosoftWordBridge"), MICROSOFT_EXCEL(
```
Comment (raphw) [inline documents4j-transformer/src/main/java/com/documents4j/conversion/ExternalConverterDiscovery.java:14]: Please format by one constant per line.

File: documents4j-transformer/src/main/java/com/documents4j/conversion/ExternalConverterDiscovery.java [hunk 8]
```diff
@@ -11,43 +11,42 @@
...
+            "com.documents4j.conversion.msoffice.MicrosoftExcelBridge"), MICROSOFT_POWERPOINT(
+                    "com.documents4j.conversion.msoffice.MicrosoftPowerpointBridge");
 
     private final String className;
 
     private ExternalConverterDiscovery(String className) {
         this.className = className;
     }
 
-    private static IExternalConverter make(Class<? extends IExternalConverter> externalConverterClass,
```
Comment (raphw) [inline documents4j-transformer/src/main/java/com/documents4j/conversion/ExternalConverterDiscovery.java:23]: Please avoid reformatting.

File: documents4j-util-transformer-process/src/main/java/com/documents4j/conversion/AbstractExternalConverter.java [hunk 9]
```diff
@@ -67,9 +67,15 @@ protected int runNoArgumentScript(File script) {
...
             // procedure, start up processes will never be killed either.
-            return makePresetProcessExecutor()
-                    .command("cmd", "/S", "/C", doubleQuote(script.getAbsolutePath()))
+            String[] command = {"cmd", "/S", "/C", doubleQuote(script.getAbsolutePath())};
+
+            int exitCode = makePresetProcessExecutor()
+                    .command(command)
                     .execute().getExitValue();
+
+            logger.trace("Got exitcode {} for command {}", exitCode, command);
```
Comment (raphw) [inline documents4j-util-transformer-process/src/main/java/com/documents4j/conversion/AbstractExternalConverter.java:76]: `command` needs wrapping.

**Review by raphw — CHANGES_REQUESTED** [review]
Thatt's brilliant, thanks for the addition. Please revert some of the formatting changes, ideally the change set is retained at a minimum.

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java (hunk 4 above)
Comment (jgoldhammer) [reply documents4j-api/src/main/java/com/documents4j/api/DocumentType.java]: Done

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java (hunk 5 above)
Comment (jgoldhammer) [reply documents4j-api/src/main/java/com/documents4j/api/DocumentType.java:6]: Done

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java (hunk 3 above)
Comment (jgoldhammer) [reply documents4j-api/src/main/java/com/documents4j/api/DocumentType.java]: Done

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java (hunk 2 above)
Comment (jgoldhammer) [reply documents4j-api/src/main/java/com/documents4j/api/DocumentType.java]: Done

File: documents4j-api/src/main/java/com/documents4j/api/DocumentType.java (hunk 1 above)
Comment (jgoldhammer) [reply documents4j-api/src/main/java/com/documents4j/api/DocumentType.java]: Done

File: documents4j-transformer-msoffice/documents4j-transformer-msoffice-base/src/main/java/com/documents4j/conversion/msoffice/AbstractMicrosoftOfficeBridge.java (hunk 6 above)
Comment (jgoldhammer) [reply documents4j-transformer-msoffice/documents4j-transformer-msoffice-base/src/main/java/com/documents4j/conversion/msoffice/AbstractMicrosoftOfficeBridge.java]: Done

File: documents4j-transformer/src/main/java/com/documents4j/conversion/ExternalConverterDiscovery.java (hunk 7 above)
Comment (jgoldhammer) [reply documents4j-transformer/src/main/java/com/documents4j/conversion/ExternalConverterDiscovery.java]: Done

File: documents4j-util-transformer-process/src/main/java/com/documents4j/conversion/AbstractExternalConverter.java (hunk 9 above)
Comment (jgoldhammer) [reply documents4j-util-transformer-process/src/main/java/com/documents4j/conversion/AbstractExternalConverter.java]: done

**Review by jgoldhammer — COMMENTED** [review]
I have made the requested changes. Thanks for reviewing so fast!

**Comment by jgoldhammer** [comment]
Can I help with other stuff or do you need more time to get this PR merged? Or do you want to test locally first? What do you think @raphw?

Thanks Jens

**Comment by jgoldhammer** [comment]
Anything else I can do? If not, you can go ahead with more tests if you want @raphw . I would be happy to get that merged soon, but it is up to you regarding time. Thanks!

**Pull request merged**